application/models/Question.php

```diff
@@ -144,6 +144,12 @@ public function rules()
             array('qid,sid,gid,parent_qid', 'numerical', 'integerOnly' => true),
             array('qid', 'unique','message' => sprintf(gT("Question id (qid) : '%s' is already in use."), $this->qid)),// Still needed ?
             array('other', 'in', 'range' => array('Y', 'N'), 'allowEmpty' => true),
+            array('other', 'filter', 'filter' => function ($value) {
```

Comment:
I believe this shall have a bigger sign. Like a separate block of lines inside the rules() method, as for highlighting this is a filter. (which is not frequently used)

Reply:
Just separate in same function ? Can be an idea :)

Keep it at anonymous function or not ? Here , i think it's cleaner enough at anonymous function (and usage only for other)

Reply:
>> Just separate in same function ? Can be an idea :)

Yes, just as a separate block of lines inside the function

>> Keep it at anonymous function or not ? Here , i think it's cleaner enough at anonymous function (and usage only for other)

Maybe a setOther method?

Reply:
> Maybe a `setOther()` method?

Think it less clear what it's done if not in existing rule …

Reply:
> > Maybe a `setOther()` method?
> 
> Think it less clear what it's done if not in existing rule …

I meant: call setOther from the rules

Reply:
? We have 'filter' rules, then adding a function like `addRulesFilterOther` ? don't like.

I prefer to create
`addRulesSubQuestiion` and `haveRulesPrimaryQuestion` in this case.

Reply:
Check please :)

Reply:
I mean you could have like this:
```
function setOther($value) {
 if ($this->getAllowOther()) {
          return $value;
      }
      return 'N';
  }
}
```

and then from within rules
```
array('other', 'filter', 'filter' => function ($value) { return $this->SetOther($value); }
```

Reply:
> Not sure is the best, but thats what I meant originally.

Compare current filter on other and filter on theme : and tell me the simple one … I think anonymous function is clearer here.

Maybe @olleharstedt or @c-schmitz advice ?

Reply:
Just keep it as it is.
No big deal!

Reply:
vote ;) 

I'm OK to create a separate function, hard to know what is the better solution. 

Create 2 function : addSubquestionsRules and addPrimaryQuestionRules seems better to do before.

Reply:
You can also write a function that produces functions. :)

Reply:
The question is : did we really need to construct some thing more complex than 
```

        $aRules[] = array('other', 'filter', 'filter' => function ($value) {
            if ($this->getAllowOther()) {
                return $value;
            }
            return 'N';
        });
```

When we don't have a better separation of rules for subquestions and questions for example.